apps/condo/domains/resident/schema/SuggestServiceProviderService.test.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Generated by `createservice organization.SuggestServiceProviderService --type queries`
+ */
+
+const { faker } = require('@faker-js/faker')
+const { values } = require('lodash')
+
+const {
+    expectToThrowAuthenticationErrorToResult,
+    expectToThrowAccessDeniedErrorToResult,
+} = require('@open-condo/keystone/test.utils')
+
+
+const { CONTEXT_FINISHED_STATUS, CONTEXT_IN_PROGRESS_STATUS } = require('@condo/domains/acquiring/constants/context')
+const {
+    TestUtils,
+    ResidentTestMixin,
+    MeterTestMixin,
+} = require('@condo/domains/billing/utils/testSchema/testUtils')
+const { RUSSIAN_COUNTRY, DEFAULT_ENGLISH_COUNTRY } = require('@condo/domains/common/constants/countries')
+const { COLD_WATER_METER_RESOURCE_ID } = require('@condo/domains/meter/constants/constants')
+const { createTestMeterResourceOwner, updateTestMeterResourceOwner, MeterResource, MeterResourceOwner } = require('@condo/domains/meter/utils/testSchema')
+const { SERVICE_PROVIDER_TYPE } = require('@condo/domains/organization/constants/common')
+const { createTestOrganization, updateTestOrganization, generateTin } = require('@condo/domains/organization/utils/testSchema')
+const { createTestProperty, updateTestProperty } = require('@condo/domains/property/utils/testSchema')
+const { suggestProviderByTestClient } = require('@condo/domains/resident/utils/testSchema')
+
+
+const addMeterResourceOwner = async (utils) => {
+    const resource = await MeterResource.getOne(utils.clients.admin, { id: COLD_WATER_METER_RESOURCE_ID })
+    return createTestMeterResourceOwner(utils.clients.admin, utils.organization, resource, {
+        address: utils.property.address,
+    })
+}
+const removeMeterResourceOwners = async (utils) => {
+    const owners = await MeterResourceOwner.getAll(utils.clients.admin, { organization: { id: utils.organization.id } })
+    for (const owner of owners) {
+        await updateTestMeterResourceOwner(utils.clients.admin, owner.id, { deletedAt: new Date().toISOString() })
+    }
+}
+
+const generateServiceProviders = async (client, count, extraAttrs) => {
+    const serviceProviders = []
+    const attrs = {
+        type: SERVICE_PROVIDER_TYPE,
+        ...extraAttrs,
+    }
+    for (let i = 0; i < count; i++) {
+        const serviceProvider = createTestOrganization(client, attrs).then(pair => pair[0])
+        serviceProviders.push(serviceProvider)
+    }
+    return Promise.all(serviceProviders)
+}
+
+describe('SuggestServiceProviderService', () => {
+
+    let utils
+
+    beforeAll(async () => {
+        utils = new TestUtils([ResidentTestMixin, MeterTestMixin])
+        await utils.init()
+        await utils.updateOrganization({ type: SERVICE_PROVIDER_TYPE })
+    })
+
+    describe('Permissions check', () => {
```

Comment:
Please refer other domains for structuring the tests, flat list is hard to navigate

Reply:
```
describe('<Util name>`, () => {
    beforeAll(async () => {
        // prepare all actors and other staff here
    })
    describe('Execute', () => {
        test('<A> can execute', async () => {
            // ...
        })
        test('<B> can execute', async () => {
            // ...
        })
        test('<C> cannot execute', async () => {
            // ...
        })
    })
    describe('Logic', () => {
        describe('guards', () => {
            
        })
        // Other logic / return / behaviour test, grouped by common pattern / type
    })
})
```